README.md

```diff
@@ -1,119 +1,44 @@
-
-We can start with a very simple architecture:
-
-1. Store a string corresponding to each type of game object that matches the file name of the serialized data.
-2. When we want to spawn an object of that type, pass in that string.
-3. During the spawning process, deserialize that file directly into a game object.
-4. Spawn the generated game object.
-
-## More sophisticated data-driven content
+1. **Id:** A unique identifier for objects of a given class (e.g. monsters, tile types or levels). Commonly stored as a components on game entities.
+2. **Item:** An in-memory representation of all of the shared data (e.g. name, asset handles, statistics) of game objects of a given kind.
+3. **Manifest:** a Bevy `Resource` which contains a mapping from identitifiers to items.
+4. **Raw manifest:** a serialization-friendly representation of the data stored in a manifest.
 
-Now, there are several areas for improvement.
-Let's go over them now, and then tackle them one at a time:
+Data is deserialized from disk into a raw manifest, which is processed into a manifest which contains a list of all available game objects of a given class.
+That manifest is then used to spawn and look up the properties of specific kinds of game objects in your game code.
 
-1. Strings are heavy identifiers: as long as we can convert to and from a nice human-readable string, we don't care about actually storing that identifier everywhere.
-2. Constantly deserializing from disk is very slow.
-3. Some final game objects are very heavy: we want to look up the data when we need it, rather than storing thousands of copies of it for every object.
-4. We can't always translate directly from the serialized to in-game representation. This is particularly common and important when we need to reference *other* serialized content inside of our game object. It also comes up when you want to simplify the serialized representation to make it more space efficient or easier to author.
+## Why manifests rock
 
-In "problems left for future work":
+An in-memory resource where you can look up the statistics for various game objects is incredibly useful:
 
-1. Text-based file formats are easy to mess up: tools for validating these would be very useful.
-2. Text-based file formats are easy to mess up: it would be great to have designer-friendly tools for authoring them.
+1. It's super easy to spawn new game objects dynamically inside of gameplay code. Simply write a helper method once, and then spawn any object you want in a single command.
+2. Manifests offer a clear list of all objects of a given kind: great for both dev tools and in-game encyclopedias.
+3. Using manifests abstracts away messy asset loading (and unloading) code into a single consistent pattern that can grow with your project.
+4. Heavy data can be deduplicated by simply looking it up in the manifest when needed.
```

Comment:
```suggestion
2. Spawning multiple clones of the same gameplay object is safer. Since the manifest is generally read-only, you know you always have a "clean" version to clone from.
3. Manifests offer a clear list of all objects of a given kind: great for both dev tools and in-game encyclopedias.
4. Using manifests abstracts away messy asset loading (and unloading) code into a single consistent pattern that can grow with your project.
5. Heavy data can be deduplicated by simply looking it up in the manifest when needed.
```